packages/netsuite-adapter/src/adapter.ts

```diff
@@ -339,15 +338,26 @@ export default class NetsuiteAdapter implements AdapterOperations {
     }
   }
 
+  private shouldFetchWithChangesDetection(
+    withChangesDetection: boolean,
+    hasFetchTarget: boolean,
+    isFirstFetch: boolean
+  ) : boolean {
+    return !isFirstFetch && (
+      this.useChangesDetection === true
+      || withChangesDetection
+      || (hasFetchTarget && !(this.useChangesDetection === false)))
```

Comment:
nit
```suggestion
      || (hasFetchTarget && this.useChangesDetection !== false))
```
Also IMO it's worth adding a comment about the implicit decision that if `this.useChangesDetection` is undefined than the default is true.